products/permissions.py

```diff
@@ -0,0 +1,11 @@
+from rest_framework.permissions import BasePermission, SAFE_METHODS

```

Comment:
I think decorator will be better instead of permissions.
So create decorator and check the role of the User.

Use decorators on appropriate APIs.